happi/audit.py

```diff
@@ -0,0 +1,424 @@
+            args: Namespace
+        """
+        raise NotImplementedError
+
+
+class Audit(Command):
+    """
+    Audits the database for valid entries.
+    """
+    def __init__(self):
+        self.name = 'audit'
+        self.help = "Inspect a database's entries"
+        self._all_devices = set()
+        self._all_items = []
+        # 9 - NO_CODE
+        self.report_code = ReportCode(9)
+
+    def add_args(self, cmd_parser):
+        cmd_parser.add_argument(
+            "--file", help='Path to the json file (database) to be audited'
+        )
+        cmd_parser.add_argument(
+            "--extras", action='store_true', default=False,
+            help='Check specifically for extra attributes'
+        )
+
+    def run(self, args):
+        """
+        Validate the database passed in with --file option
+        """
+        if args.file is not None:
+            self.process_args(args.file, args)
+        else:
+            """
+            Validate the database defined in happi.cfg file
+            """
+            path_to_config = Client.find_config()
+
+            if path_to_config:
+                logger.info('Using Client cfg path %s ', path_to_config)
+                config = ConfigParser()
+                config.read(path_to_config)
+                try:
+                    database_path = config.get('DEFAULT', 'path')
+                except Exception as e:
+                    logger.error('Error when trying '
+                                 'to get database path %s', e)
+                    return
+                else:
+                    self.process_args(database_path, args)
+            else:
+                logger.error('Could not find the Happi Configuration file')
+                sys.exit(1)
+
+    def process_args(self, database_path, args):
+        """
+        Checks to see if a valid path to database was provided.
+        If --extras is provided, will call for check xtra attributes, otherwise
+        it will just proceed with parsing and validating the database call.
+        """
+        if self.validate_file(database_path):
+            logger.info('Using database file at %s ', database_path)
+            if args.extras:
+                self.check_extra_attributes(database_path)
+            else:
+                self.parse_database(database_path)
+        else:
+            logger.error('The database %s file path '
+                         'could not be validated', database_path)
+            sys.exit(1)
+
+    def validate_file(self, file_path):
+        """
+        Tests whether a path is a regular file
+
+        Parameters
+        -----------
+        file_path: str
+            File path to be validate
+
+        Returns
+        -------
+        bool
+            To indicate whether the file is a valid regular file
+        """
+        return os.path.isfile(file_path)
+
+    def validate_container(self, item):
+        """
+        Validates container definition
```

Comment:
`item` should be in the `Parameters` docstring here and below

Reply:
Ok, added this in 1f87a91